src/js/navIcons.js

```diff
@@ -30,17 +30,20 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
             show: "gpii-fd-show"
         },
         modelListeners: {
-            "isActive.setState": {
-                listener: "gpii.firstDiscovery.icon.setState",
-                args: ["{that}", "{change}.value", "{change}.oldValue"]
+            "isActive.setActiveState": {
+                "this": "{that}.container",
+                method: "toggleClass",
+                args: ["{that}.options.styles.active", "{change}.value"]
+            },
+            "isVisited.setVisitedState": {
+                listener: "gpii.firstDiscovery.icon.setVisitedState",
+                args: ["{that}", "{change}.value"]
             }
         }
     });
 
-    gpii.firstDiscovery.icon.setState = function (that, isActive, isActivePrev) {
-        that.container.toggleClass(that.options.styles.active, isActive);
-
-        if (isActivePrev && !isActive) {
+    gpii.firstDiscovery.icon.setVisitedState = function (that, isVisited) {
+        if (isVisited) {
             that.locate("doneIndicator").addClass(that.options.styles.show);
```

Comment:
@cindyli maybe not "justVisited" as it should remain after having pressed next. Maybe something more along the lines of a decision was made. Like "confirmed" or something.